### PR TITLE
Remove unnecessary lock in MManager

### PR DESCRIPTION
**Pull request by SilverNarcissus**
There is an unnecessary lock in MManager. I have removed this in this PR.

### CODE REVIEW

**Comment by HTHou** [comment]
Should we consider to submit a PR to rel/0.10 branch as well?

File: server/src/main/java/org/apache/iotdb/db/metadata/MManager.java [hunk 1]
```diff
@@ -1121,7 +970,6 @@ public StorageGroupMNode getStorageGroupNodeByPath(PartialPath path) throws Meta
    */
   public MNode getDeviceNodeWithAutoCreateAndReadLock(
       PartialPath path, boolean autoCreateSchema, int sgLevel) throws MetadataException {
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/metadata/MManager.java:972]: If there is no read lock anymore, what about change the method name to `getDeviceNodeWithAutoCreate`?

File: server/src/main/java/org/apache/iotdb/db/metadata/MManager.java [hunk 2]
```diff
@@ -1205,25 +1039,16 @@ public String getDeviceId(PartialPath path) {
     return device;
   }
 
+  // concurrent warning: you should get parent read lock before calling this method
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/metadata/MManager.java:1042]: Remove it?

File: server/src/main/java/org/apache/iotdb/db/metadata/MManager.java (hunk 1 above)
Comment (qiaojialin) [reply server/src/main/java/org/apache/iotdb/db/metadata/MManager.java:970]: fixed

File: server/src/main/java/org/apache/iotdb/db/metadata/MManager.java (hunk 2 above)
Comment (qiaojialin) [reply server/src/main/java/org/apache/iotdb/db/metadata/MManager.java]: fixed

**Review by qiaojialin — APPROVED** [review]

**Review by HTHou — APPROVED** [review]

**Pull request merged**